bpf/bpf_host.c

```diff
@@ -1148,13 +1148,13 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 					      from_host);
 		ctx_store_meta(ctx, CB_SRC_LABEL, identity);
 		if (from_host) {
-# if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE)
+# if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV4)
 			/* If we don't rely on BPF-based masquerading, we need
 			 * to pass the srcid from ipcache to host firewall. See
 			 * comment in ipv4_host_policy_egress() for details.
 			 */
 			ctx_store_meta(ctx, CB_IPCACHE_SRC_LABEL, ipcache_srcid);
-# endif
+# endif /* defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV4) */
```

Comment:
The asymmetry with the IPv6 path concerns me. But that's not something introduced by your PR :).

Reply:
What do you mean, having two different paths `ENABLE_MASQUERADE_IPV4` and `...IPV6`? I can't see how to avoid it with the options supported by the Agent, given that you can turn on/off masquerading for one IP version or the other independently :thinking:.

Reply:
It looks like we have special handling for the "no IPv4 BPF masq" case in HostFW, but not yet in the IPv6 path.

I haven't tracked it down further at all, but would speculate that this is because the corresponding IPv6 HostFW path right now only has the "no BPF masq" handling. And would need to be extended in similar fashion as IPv4 to also handle BPF masq.

Needs further digging whether that handling for BPF masq in HostFW is an optional optimization, or strictly required.

Reply:
OK, right, let's disable BPF IPv6 masquerading in conjunction with Host Firewall until we figure out the details here. Thanks a lot for the review and feedback!

Reply:
Thanks! For the record, this is tracked in https://github.com/cilium/cilium/issues/26074.